### DISCUSSION TITLE
I have a json, I can't output it to seq as an entity, pls help.

### DISCUSSION THREAD

**Discussion by xiaoyutx94** [datalust/seq-tickets#1815] [Help] [answered]
JSON e.g.
`"{\"id\":null,\"content\":{\"uangMukaDPP\":0.0,\"uangMukaPPN\":0.0,\"uangMukaPPnBM\":0.0,\"jumlahDPP\":149.0000000000,\"jumlahPPN\":151.0000000000,\"jumlahPPnBM\":0.0,\"fgUangMuka\":0,\"fgPengganti\":\"0\",\"referensi\":\"0087HP_SD/CIKARANG\",\"tanggalFaktur\":\"2023-03-04\",\"masa\":3,\"tahun\":2023,\"kdTransaksi\":\"01\",\"keterangan\":\"\",\"idKeteranganTambahan\":0,\"nomorFaktur\":\"1112096472\",\"npwpPembeli\":\"11414000\",\"nikPembeli\":null,\"namaPembeli\":\"aNDIRI\",\"alamatPembeli\":\"a 05\"},\"wpId\":\"5\",\"branchCode\":\"ALL\",\"objekFakturs\":[{\"no\":1,\"item\":{\"diskon\":0.0000000000,\"hargaSatuan\":1702.7000000000,\"hargaTotal\":1701.7000000000,\"jumlahBarang\":1.0000000000,\"keterangan\":null,\"kodeObjek\":\"4+64-BLACK\",\"namaObjek\":\"4+64 BLACK\"},\"tax\":{\"dpp\":1102.7000000000,\"ppn\":197.3000000000,\"ppnbm\":0.0,\"tarifPPnBM\":0.0}},{\"no\":2,\"item\":{\"diskon\":0.0000000000,\"hargaSatuan\":1945.9500000000,\"hargaTotal\":1946.3000000000,\"jumlahBarang\":1.0000000000,\"keterangan\":null,\"kodeObjek\":\"4+64-GREEN\",\"namaObjek\":\"4+64 G GREEN\"},\"tax\":{\"dpp\":946.3000000000,\"ppn\":253.7000000000,\"ppnbm\":0.0,\"tarifPPnBM\":0.0}}],\"status\":null}"`

`JObject data = "{\"id\":null,\"content\":{\"uangMukaDPP\":0.0,\"uangMukaPPN\":0.0,\"uangMukaPPnBM\":0.0,\"jumlahDPP\":149.0000000000,\"jumlahPPN\":151.0000000000,\"jumlahPPnBM\":0.0,\"fgUangMuka\":0,\"fgPengganti\":\"0\",\"referensi\":\"0087HP_SD/CIKARANG\",\"tanggalFaktur\":\"2023-03-04\",\"masa\":3,\"tahun\":2023,\"kdTransaksi\":\"01\",\"keterangan\":\"\",\"idKeteranganTambahan\":0,\"nomorFaktur\":\"1112096472\",\"npwpPembeli\":\"11414000\",\"nikPembeli\":null,\"namaPembeli\":\"aNDIRI\",\"alamatPembeli\":\"a 05\"},\"wpId\":\"5\",\"branchCode\":\"ALL\",\"objekFakturs\":[{\"no\":1,\"item\":{\"diskon\":0.0000000000,\"hargaSatuan\":1702.7000000000,\"hargaTotal\":1701.7000000000,\"jumlahBarang\":1.0000000000,\"keterangan\":null,\"kodeObjek\":\"4+64-BLACK\",\"namaObjek\":\"4+64 BLACK\"},\"tax\":{\"dpp\":1102.7000000000,\"ppn\":197.3000000000,\"ppnbm\":0.0,\"tarifPPnBM\":0.0}},{\"no\":2,\"item\":{\"diskon\":0.0000000000,\"hargaSatuan\":1945.9500000000,\"hargaTotal\":1946.3000000000,\"jumlahBarang\":1.0000000000,\"keterangan\":null,\"kodeObjek\":\"4+64-GREEN\",\"namaObjek\":\"4+64 G GREEN\"},\"tax\":{\"dpp\":946.3000000000,\"ppn\":253.7000000000,\"ppnbm\":0.0,\"tarifPPnBM\":0.0}}],\"status\":null}".ToObject<JObject>();

logger.Info("Status Code：OK；RequestData：{@RequestData}",data);`

can not to output to seq.

**Comment by KodrAus** [comment]
Hi @snailg :wave:

You may need to use [`Destructurama.JsonNet`](https://github.com/destructurama/json-net) before Serilog will destructure that object. Hope that helps!

**Reply by xiaoyutx94** [reply]
sorry，I didn't use Serilog, I used nlog.

**Reply by nblumhardt** [reply, marked as answer]
Hi @snailg,

This seems to discuss the kind of feature you're looking for in NLog:

https://github.com/datalust/seq-extensions-logging#including-literal-json-strings-in-log-events

I'm not sure how to adapt it to your scenario, but referencing it in a new Stack Overflow post with all the details and the `nlog` tag will very likely get the right eyes on it. It'd be great to hear how you go!

**Reply by xiaoyutx94** [reply]
> Hi @snailg,
> 
> This seems to discuss the kind of feature you're looking for in NLog:
> 
> https://github.com/datalust/seq-extensions-logging#including-literal-json-strings-in-log-events
> 
> I'm not sure how to adapt it to your scenario, but referencing it in a new Stack Overflow post with all the details and the `nlog` tag will very likely get the right eyes on it. It'd be great to hear how you go!

Thank you very much！
